scalding-parquet/src/main/scala/com/twitter/scalding/parquet/tuple/macros/impl/ParquetTupleConverterProvider.scala

```diff
@@ -0,0 +1,89 @@
+package com.twitter.scalding.parquet.tuple.macros.impl
+
+import com.twitter.bijection.macros.impl.IsCaseClassImpl
+import com.twitter.scalding.parquet.tuple.scheme.ParquetTupleConverter
+
+import scala.reflect.macros.Context
+
+object ParquetTupleConverterProvider {
+  def toParquetTupleConverterImpl[T](ctx: Context)(implicit T: ctx.WeakTypeTag[T]): ctx.Expr[ParquetTupleConverter] = {
+    import ctx.universe._
+
+    if (!IsCaseClassImpl.isCaseClassType(ctx)(T.tpe))
+      ctx.abort(ctx.enclosingPosition,
+        s"""We cannot enforce ${T.tpe} is a case class,
+            either it is not a case class or this macro call is possibly enclosed in a class.
+            This will mean the macro is operating on a non-resolved type.""")
+
+    def buildGroupConverter(tpe: Type, parentTree: Tree, isOption: Boolean, idx: Int, converterBodyTree: Tree): Tree = {
+      q"""new _root_.com.twitter.scalding.parquet.tuple.scheme.ParquetTupleConverter($parentTree, $isOption, $idx){
+            override def newConverter(i: Int): _root_.parquet.io.api.Converter = {
+              $converterBodyTree
+              throw new RuntimeException("invalid index: " + i)
+            }
+
+            override def createValue(): Any = {
+              if(fieldValues.isEmpty) null
+              else classOf[$tpe].getConstructors()(0).newInstance(fieldValues.toSeq.map(_.asInstanceOf[AnyRef]): _*)
```

Comment:
we should be able to do better than creating through reflection. After all it's a macro.